diesel_cli/src/infer_schema_internals/information_schema.rs

```diff
@@ -175,23 +176,25 @@ where
 {
     use self::information_schema::tables::dsl::*;
 
-    let default_schema = Conn::Backend::default_schema(connection)?;
-    let schema_name = match schema_name {
-        Some(name) => name,
-        None => &default_schema,
+    let db_schema_name = match schema_name {
+        Some(name) => Cow::Borrowed(name),
+        None => Cow::Owned(Conn::Backend::default_schema(connection)?),
     };
 
     let mut table_names = tables
-        .select((table_name, table_schema))
-        .filter(table_schema.eq(schema_name))
+        .select(table_name)
+        .filter(table_schema.eq(db_schema_name.as_ref()))
         .filter(table_name.not_like("\\_\\_%"))
         .filter(table_type.like("BASE TABLE"))
-        .order(table_name)
-        .load::<TableName>(connection)?;
-    for table in &mut table_names {
-        table.strip_schema_if_matches(&default_schema);
-    }
-    Ok(table_names)
+        .load::<String>(connection)?;
+    table_names.sort_unstable();
```

Comment:
If we care about order here we should use `sort` instead of `sort_unstable`.
Additionally why not just use `.sort_by_key()` and then return the table name there instead of doing that dance with the schema name below?

Reply:
> If we care about order here we should use sort instead of sort_unstable.

There are only two differences between a [stable sort](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort) and an [unstable sort](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_unstable):
- The stable sort guarantees the non-reordering of equal elements
- The unstable sort is usually fastest

But this query is not supposed to return any duplicates, so there will never be equal elements.
Besides, since here the comparison function is the default, that is that of two strings being equal, even if there were duplicates, it wouldn't change anything to what happens afterwards. 

Using stable sort instead of unstable sort may have an effect when using `sort_by` or `sort_by_key` in very specific scenarios where you need to preserve the order of initial elements in case two elements are equal, but in the general case unstable sort is strictly better.

We are not in such a case, which is why I favored the unstable sort.

> Why not just use .sort_by_key() and then return the table name there instead of doing that dance with the schema name below?

Because it took me more time than I would have liked to understand exactly what was happening because what the code was doing is:
- Get the name of the default schema
- Query the table name AND the schema name with a constraint on the schema name being something we already know
- Post-patch the schema name so that if we initially queried None which led to the schema name being that of the default schema we get None back in the table name (and doing so by checking every `TableName` even though they could only all match or all not match)

Though what we want to do is simply:
- Query the table names for a specific schema
- Create objects that pack both the table name and the schema name that we asked for.

The previous approach raised questions like:
- Why is the check on whether we should erase the schema made on all the tables one by one ? Do they have different values ?
- Why do we query the schema name with the table name ? Is there some weird case somewhere where it could be different from what we called the function with ?

Overall I feel like this new approcach is much more straightforward hence easier to understand and feels much less as a dance to me than the previous one. (And as a bonus may dodge a query on MySQL backends).

Reply:
I have just realized while going to bed that there's actually a difference between my implementation and the previous one: if the argument is `Some(default_schema_name)` it used to be turned into `None`, while my implementation didn't do that.

I've updated the code to preserve that previous behaviour, still in a more explicit way than used to be.

However the systematic use of `strip_schema_if_matches` on the database output also guaranteed that behaviour so there was more to it than I originally thought, so I cannot advocate for my implementation as much as I did. However if we revert this to the previous implementation, we should add some documentation about why it's there because it was non-obvious enough that I went and rewrote it.

Reply:
>      There are only two differences between a stable sort and an unstable sort:
>>    The stable sort guarantees the non-reordering of equal elements
>>   The unstable sort is usually the fastest
> 
> But this query is not supposed to return any duplicates, so there will never be equal elements.
> Besides, since here the comparison function is the default, that is that of two strings being equal, even if there were     duplicates, it wouldn't change anything to what happens afterwards.

> Using stable sort instead of unstable sort may have an effect when using sort_by or sort_by_key in very specific scenarios where you need to preserve the order of initial elements in case two elements are equal, but in the general case unstable sort is strictly better.

> We are not in such a specific case, which is why I favored the unstable sort.

Right, forgot that `table` names should be unique here 🤦‍

> Overall I feel like this new approcach is much more straightforward hence easier to understand and feels much less as a dance to me than the previous one. (And as a bonus may dodge a query on MySQL backends).

Normally we try to leave old code in place because `git blame` then points at the correct commits that have introduced that code and contain that contain additional information about the implementation. In this case the change seems fine because it actually improves the implementation quite significantly.